indexer/node/fetcher.go

```diff
@@ -0,0 +1,63 @@
+package node
+
+import (
+	"math/big"
+
+	"github.com/ethereum/go-ethereum/core/types"
+)
+
+const (
+	// Max number of headers that's bee returned by the Fetcher at once. This will
+	// eventually be configurable
+	maxHeaderBatchSize = 50
+)
+
+type Fetcher struct {
+	ethClient EthClient
+
+	// TODO: Store the last header block hash to ensure
+	// the next batch of headers builds on top
+	nextStartingBlockHeight *big.Int
```

Comment:
This could be a regular number I think if you prefer.   I personally like just using big.Int without thinking too hard about it though.

Reply:
I believe the number in the header is a `big.Int` so it makes sense to use the same type. `big.Int` is a weird type its one of my least fav parts of go, takes awhile to get used to

Reply:
+1 I also hate using big int but makes sense to avoid future problems if we always directly convert to `uint64`